src/main.rs

```diff
@@ -1585,14 +1585,18 @@ async fn main() {
     if setup.enable_discovery {
         let device_id = setup.session_config.device_id.clone();
 
-        discovery = Some(
-            librespot::discovery::Discovery::builder(device_id)
-                .name(setup.connect_config.name.clone())
-                .device_type(setup.connect_config.device_type)
-                .port(setup.zeroconf_port)
-                .launch()
-                .unwrap(),
-        );
+        discovery = match librespot::discovery::Discovery::builder(device_id)
+            .name(setup.connect_config.name.clone())
+            .device_type(setup.connect_config.device_type)
+            .port(setup.zeroconf_port)
+            .launch()
+        {
+            Ok(d) => Some(d),
+            Err(e) => {
+                warn!("{}", e);
```

Comment:
This probably warrants error level instead of warning level.

Reply:
The question becomes do you want to cut and run there or continue?

Reply:
It's not necessarily fatal so long as there are cached creds.

Reply:
It's clearly an error when we want to set up a mdns service and that fails.

Reply:
We would need to check if we have cached creds then if we don't have cached creds it's fatal.

It would be the equivalent of disabling discovery and not having any creds.

Reply:
The intention here is to launch a zeroconf service. Not to launch a zeroconf service if and only if there are no credentials. That's the other way around, and not what is implied here.

Reply:
To be clear I'm not saying we should exit immediately (that's where cached credentials can come in to save the day) but it's clearly an error and not a warning.

Reply:
With regards to exiting or not: on the other hand, it is also no guarantee that the user wants to reuse cached credentials instead of discovery. It can go both ways. Personally I think credential caching doesn't make much sense together with discovery.

Reply:
I agree. Unless someone disables discovery they would expect it.

It's now:

```rust
            Ok(d) => Some(d),
            Err(e) => {
                error!("Discovery Error: {}", e);
                exit(1);
            }
            
```

Reply:
You know my feelings on cred caching. I'm not a fan at all.